SCons/Taskmaster/__init__.py

```diff
@@ -604,9 +608,59 @@ def __init__(self, targets=[], tasker=None, order=None, trace=None):
             order = lambda l: l
         self.order = order
         self.message = None
-        self.trace = trace
         self.next_candidate = self.find_next_candidate
         self.pending_children = set()
+        self.trace = False
+        self.configure_trace(trace)
+
+    def configure_trace(self, trace=None):
+        """
+        This handles the command line option --taskmastertrace=
+        It can be:
+        -           : output to stdout
+        <filename>  : output to a file
+        False/None  : Do not trace
+        """
+        if not trace:
+            self.trace = False
+            return
+
+        # TODO: May want to switch format to something like this.
+        # log_format = (
+        #     '%(relativeCreated)05dms'
+        #     ':%(relfilename)s'
+        #     ':%(funcName)s'
+        #     '#%(lineno)s'
+        #     ': %(message)s'
+        # )
+        tm_formatter = logging.Formatter('Taskmaster: %(message)s')
+        if isinstance(trace, io.StringIO):
+            log_handler = logging.StreamHandler(trace)
+        elif trace == '-':
+            log_handler = logging.StreamHandler(sys.stdout)
+        elif trace:
+            log_handler = logging.FileHandler(filename=trace)
+
+        logger = logging.getLogger('Taskmaster')
+        logger.setLevel(level=logging.DEBUG)
+        logger.addHandler(log_handler)
+        self.trace = logger
+
+        logger.log_handler = log_handler
+
+        # Now setup Task's logger.
+        tl = logging.getLogger("Task")
+        tl.setLevel(level=logging.DEBUG)
+        tl.addHandler(log_handler)
+        task_formatter = logging.Formatter('%(name)s.%(message)s')
+        Task.LOGGER = tl
+
+        log_handler.setFormatter(DispatchingFormatter({
```

Comment:
This came out weirdly formatted - might as well use the Black style since it's new code.  I'm going to suggest kwargs for readability, too, but certainly not essential. Like: 
```py
        log_handler.setFormatter(
            DispatchingFormatter(
                formatters = {
                    'Taskmaster': tm_formatter,
                    'Task': task_formatter,
                },  
                default_formatter = logging.Formatter('%(message)s'),
            )
        )
```

Reply:
Good catch changed to this. Much easier to read.